modules/MusicDownloader.js

```diff
@@ -86,24 +127,33 @@ async function downloadAlbum(url) {
         clientSecret: process.env.SPOTIFY_CLIENT_SECRET,
     });
 
-    const data = await spotify.getAlbum(url);
-    const dir = sterylizator(data.name);
-    if (!fs.existsSync(`./mp3/onDemand/${dir}`)){
-        fs.mkdirSync(`./mp3/onDemand/${dir}`);
-    }
-    for (let i in data.tracks) {
-        let song = await getTrackInfo(data.tracks[i])
-        logger('log',`Pobieranie: ${song.name+' by: '+song.artists.join(', ')}`,'downloadAlbum');
-    }
-    const album = await spotify.downloadAlbum(url);
-    for (let i in album) {
-        let song = await getTrackInfo(data.tracks[i]);
-        const file = sterylizator(song.artists.join('-')+'_'+song.name);
-        if (fs.existsSync(`./mp3/onDemand/${dir}/${file}.mp3`)) {
-            logger('warn',`${file}.mp3 - Plik istnieje!`,'downloadAlbum');
-            continue;
+    try {
+        const data = await spotify.getAlbum(url);
+        const dir = sterylizator(data.name);
+        if (!fs.existsSync(`./mp3/onDemand/${dir}`)) {
+            fs.mkdirSync(`./mp3/onDemand/${dir}`);
+        }
+        for (let i in data.tracks) {
+            let song = await getTrackInfo(data.tracks[i])
+            logger('log', `Pobieranie: ${song.name + ' by: ' + song.artists.join(', ')}`, 'downloadAlbum');
+        }
+        const album = await spotify.downloadAlbum(url);
+        for (let i in album) {
+            let song = await getTrackInfo(data.tracks[i]);
+            const file = sterylizator(song.artists.join('-') + '_' + song.name);
+            if (fs.existsSync(`./mp3/onDemand/${dir}/${file}.mp3`)) {
+                logger('warn', `${file}.mp3 - Plik istnieje!`, 'downloadAlbum');
+                continue;
+            }
+            fs.writeFileSync(`./mp3/onDemand/${dir}/${file}.mp3`, album[i]);
+        }
+    } catch (e) {
+        logger('error', "Błąd w trakcie wykonywania funkcji downloadSong", 'downloadAlbum');
+        logger('error', e, 'downloadAlbum');
+        if (global.debugmode === true) {
+            DebugSaveToFile('MusicDownloader', 'downloadAlbum', 'error_main_function', e);
```

Comment:
Zmień error_main_function na catched_error
```suggestion
            DebugSaveToFile('MusicDownloader', 'downloadAlbum', 'catched_error', e);
```